extra_foam/gui/windows/pump_probe_w.py

```diff
@@ -138,8 +148,61 @@ def initUI(self):
 
     def initConnections(self):
         """Override."""
-        pass
+        self._ctrl_widget.save_btn.clicked.connect(self.saveToFile)
 
     def closeEvent(self, QCloseEvent):
         self._ctrl_widget.resetAnalysisType()
         super().closeEvent(QCloseEvent)
+
+    def saveToFile(self):
+        # Get the current data
+        if not len(self._queue):
+            return
+        data = self._queue[0]
+
+        # Open file dialog
+        filepath, filter = QFileDialog.getSaveFileName(
+            caption="Save image",
+            directory=osp.expanduser("~"),
+            filter=f"{SaveFile.NPZ.value};;{SaveFile.TXT.value}")
+        filter = SaveFile(filter)
+
+        # Validate filepath
+        if not filepath:
+            return
+        suffix = f".{filter.name.lower()}"
+        if not filepath.lower().endswith(suffix):
+            filepath += suffix
+
+        # Copy reference file from tmp folder to desired destination
+        os.makedirs(osp.dirname(filepath), exist_ok=True)
```

Comment:
This comment doesn't seem to match the code? And why is it necessary to use `os.makedirs()` here? I thought we could trust that any directory returned by `QFileDialog.getSaveFileName()` already exists since the user would need to create it first in the dialog.